### PR TITLE
Compressed input for bowtie2

### PR DESCRIPTION
**Pull request by dpryan79**
Xref: https://github.com/galaxyproject/tools-iuc/issues/1128

This adds support of fastqsanger.gz/fastqsanger.bz2 input/output in bowtie2.

### CODE REVIEW

File: tools/bowtie2/bowtie2_wrapper.xml [hunk 1]
```diff
@@ -19,6 +19,66 @@
...
+            #if $library.input_1.forward.is_of_type("fastq.gz", "fastqsanger.gz"):
+                #set read1 = "input_f.fastq.gz"
+                #set compressed = "GZ"
+            #else if $library.input_1.forward.is_of_type("fastq.bz2", "fastqsanger.bz2"):
+                #set read1 = "input_f.fastq.bz2"
+                #set compressed = "BZ2"
+            #else:
+                #set read1 = "input_f.fastq"
+            #end if
+            ln -s "${library.input_1.forward}" "${read1}" &&
```
Comment (bgruening) [inline tools/bowtie2/bowtie2_wrapper.xml]: single quote, also above

File: tools/bowtie2/bowtie2_wrapper.xml [hunk 2]
```diff
@@ -19,6 +19,66 @@
...
+            #if $library.input_1.reverse.is_of_type("fastq.gz", "fastqsanger.gz"):
+                #set read2 = "input_r.fastq.gz"
+                #set compressed = "GZ"
+            #else if $library.input_1.reverse.is_of_type("fastq.bz2", "fastqsanger.bz2"):
+                #set read2 = "input_r.fastq.bz2"
+                #set compressed = "BZ2"
+            #else:
+                #set read2 = "input_r.fastq"
+            #end if
+            ln -s "${library.input_1.reverse}" "${read2}" &&
```
Comment (bgruening) [inline tools/bowtie2/bowtie2_wrapper.xml]: quotes

File: tools/bowtie2/bowtie2_wrapper.xml [hunk 3]
```diff
@@ -19,6 +19,66 @@
...
+            #if $library.input_1.is_of_type("fastq.gz", "fastqsanger.gz"):
+                #set read1 = "input_f.fastq.gz"
+                #set compressed = "GZ"
+            #else if $library.input_1.is_of_type("fastq.bz2", "fastqsanger.bz2"):
+                #set read1 = "input_f.fastq.bz2"
+                #set compressed = "BZ2"
+            #else:
+                #set read1 = "input_f.fastq"
+            #end if
+            ln -s "${library.input_1}" "${read1}" &&
```
Comment (bgruening) [inline tools/bowtie2/bowtie2_wrapper.xml]: quotes

File: tools/bowtie2/bowtie2_wrapper.xml [hunk 4]
```diff
@@ -72,7 +142,10 @@
                 ${library.paired_options.no_overlap}
             #end if
             #if str( $library.unaligned_file ) == "true":
-                --un-conc '$output_unaligned_reads_l'
+                --un-conc $output_unaligned_reads_l
```
Comment (bgruening) [inline tools/bowtie2/bowtie2_wrapper.xml]: Quotes missing?

File: tools/bowtie2/bowtie2_wrapper.xml [hunk 5]
```diff
@@ -209,15 +282,15 @@
...
-                <param name="input_1" format="fastqsanger" type="data" label="FASTQ file" help="Must be of datatype &quot;fastqsanger&quot;" />
-                <param name="unaligned_file" type="boolean" truevalue="true" falsevalue="false" checked="False" label="Write unaligned reads (in fastq format) to separate file(s)" help="--un/--un-conc; This triggers --un parameter for single reads and --un-conc for paired reads" />
-                <param name="aligned_file" type="boolean" truevalue="true" falsevalue="false" checked="False" label="Write aligned reads (in fastq format) to separate file(s)" help="--al/--al-conc; This triggers --al parameter for single reads and --al-conc for paired reads" />
+                <param name="input_1" format="fastqsanger,fastqsanger.gz,fastqsanger.bz2" type="data" label="FASTQ file" help="Must be of datatype &quot;fastqsanger&quot;" />
+                <param name="unaligned_file" type="boolean" truevalue="true" falsevalue="false" checked="False" label="Write unaligned reads (in fastq format) to separate file(s)" help="--un/--un-conc (possibly with -gz or -bz2); This triggers --un parameter for single reads and --un-conc for paired reads" />
```
Comment (bgruening) [inline tools/bowtie2/bowtie2_wrapper.xml]: macro?

File: tools/bowtie2/bowtie2_wrapper.xml (hunk 4 above)
Comment (dpryan79) [inline tools/bowtie2/bowtie2_wrapper.xml]: That one should have been deleted, in fact.

File: tools/bowtie2/bowtie2_wrapper.xml (hunk 5 above)
Comment (dpryan79) [inline tools/bowtie2/bowtie2_wrapper.xml]: Possibly, they end up being slightly different subsets of options for SE/PE/PE collection.

**Comment by bgruening** [comment]
Thanks @dpryan79!

**Pull request merged**